packages/lib-core/src/extensions/actions.ts

```diff
@@ -0,0 +1,119 @@
+import type { ReactNode } from 'react';
+import type { ExtensionHook, ExtensionK8sKindVersionModel } from '../api/common-types';
+import type { CodeRef, Extension } from '../types/extension';
+import type { AccessReviewResourceAttributes } from './core-types';
+
+/** ActionProvider contributes a hook that returns list of actions for specific context */
+export type ActionProvider = Extension<
+  'core.action/provider',
```

Comment:
@andrewballantyne We should consider writing a custom ESLint rule that checks all specific extension types and ensures the first generic type parameter is a string literal that starts with `core.` prefix.

https://typescript-eslint.io/docs/development/custom-rules/